autometa/config/databases.py

```diff
@@ -66,230 +97,504 @@ class Databases:
+        Parameters
+        ----------
+        section : str
+            section to retrieve for `checksums` section.
+            Choices include: 'ncbi' and 'markers'.
+        compare_checksums : bool, optional
+            Also check if database information is up-to-date with current hosted databases.
+            (default is False).
 
         Returns
         -------
-        NoneType
+        bool
+            True if all database dependencies are satisfied, otherwise False.
 
         """
-        if not self.config.has_section('databases'):
-            self.config.add_section('databases')
-        for section in Databases.SECTIONS:
-            if self.config.has_option('databases', section):
-                continue
-            outdir = DEFAULT_CONFIG.get('databases', section)
-            self.config.set('databases', section, outdir)
+        any_missing = self.get_missing(section=section, validate=True)
+        if compare_checksums:
+            any_invalid = self.compare_checksums(section=section, validate=True)
+        else:
+            any_invalid = False
+        if not any_missing and not any_invalid:
+            return True
+        else:
+            return False
+
+    def get_remote_checksum(self, section, option):
+        """Get the checksum from provided `section` respective to `option` in `self.config`.
+
+        Parameters
+        ----------
+        section : str
+            section to retrieve for `checksums` section.
+            Choices include: 'ncbi' and 'markers'.
+        option : str
+            `option` in `checksums` section corresponding to the section checksum file.
+
+        Returns
+        -------
+        str
+            checksum of remote md5 file. e.g. 'hash filename\n'
+
+        Raises
+        -------
+        ConnectionError
+            Failed to connect to host for provided `option`.
+
+        """
+        if section not in {'ncbi','markers'}:
+            raise ValueError(f'"section" must be "ncbi" or "markers". Provided: {section}')
+        if section == 'ncbi':
+            host = self.config.get(section, 'host')
+            ftp_fullpath = self.config.get('checksums', option)
+            chksum_fpath = ftp_fullpath.split(host)[-1]
+            with FTP(host) as ftp, tempfile.TemporaryFile() as fp:
+                ftp.login()
+                result = ftp.retrbinary(f'RETR {chksum_fpath}', fp.write)
+                if not result.startswith('226 Transfer complete'):
+                    raise ConnectionError(f'{chksum_fpath} download failed')
+                ftp.quit()
+                fp.seek(0)
+                checksum = fp.read().decode()
+        elif section == 'markers':
+            url = self.config.get('checksums', option)
+            with requests.Session() as session:
+                resp = session.get(url)
+                if not resp.ok:
+                    raise ConnectionError(f'Failed to retrieve {url}')
+                checksum = resp.text
```

Comment:
If the user supplies their own marker databases will this cause an error? The code looks different to the above, I guess assuming that the URL is not an FTP site. What if the user supplies a local file or a FTP address and/or doesn't provide a checksum entry in their config?

Reply:
I think the user could add their databases within `checksums` as well as the `markers` section where the `option` should match in both.

Reply:
This is by default retrieving the md5 over http requests and not FTP. So far there is no mechanism in place to determine this based on the URL